### DISCUSSION TITLE
EPS with multi-clipping paths conversion to PNG

### DISCUSSION THREAD

**Discussion by jetpack1116** [ImageMagick/ImageMagick#4169] [Help]
ImageMagick cannot know which is the main clipping path of an EPS image with multiple paths.
So I cannnot automate the conversion of EPS to PNG because it shows invalid result. 
ImageMagick only considers the first path as the main clipping path of an EPS image.
What if the main clipping path is the 2nd or 3rd path.
So basically the result will be wrong.

To this regard, I plan to use GhostScript to convert the EPS image to MIFF.
GhostScript can recognize the correct clipping path.
From this MIFF file, I will use ImageMagick to convert MIFF to PNG or other raster formats.

Problem is, standard GhostScript installation does not include the device to convert to MIFF.

Are there any other image format that is similar to MIFF that can be used as intermediary format to be used later with ImageMagick to convert to PNG, JPEG, or other raster image formats?

**Comment by fmw42** [comment]
MIFF is an ImageMagick proprietary format.  Ghostscript knows nothing about MIFF. Perhaps you want TIFF?

**Reply by jetpack1116** [reply]
@fmw42 

There is a device in GhostScript that can write MIFF.
That device is `miff24`.

But it is not included in standard GhostScript.
To use that device, we need to recompile GhstScript to include that device.

TIFF format is good candidate.
Is it similar or just near similar to MIFF? 
Fast lightweight, suitable as intermediate format for conversion and can be used by ImageMagick?
i just need your insights guys.
Thanks.

**Reply by fmw42** [reply]
I do not know if miff24 is the same as ImageMagick miff:. I doubt it.  I only recommend TIFF because it can deal with clip paths and transparency and layers.  It is not anything similar to miff:.

**Reply by jetpack1116** [reply]
@fmw42 
Ok, noted about TIFF...I will try using it.

**Reply by jetpack1116** [reply]
I ended up using `png` with device `pngalpha`
This is the only format that maintains transparency

generated TIFF have white backgrounds

**Comment by snibgo** [comment]
@jetpack1116: please link to a EPS file with multiple paths.

How do you know which path is the "main clipping path"?

As far as I know, IM knows nothing about clipping paths that may be present in an EPS. I don't understand how IM (which calls GS) uses the wrong clipping path, but using GS directly gets the correct clipping path.

If a simple GS command gives the result you want, you could put that in your delegates.xml file. Use a name that IM doesn't use, such as "MYEPS". Then you can convert an EPS file like this:
```
magick MYEPS:myfile.eps out.png
```

**Reply by jetpack1116** [reply]
@snibgo 

thanks for this tip.

is it `magick` only?
or is it  `magic convert` and then I can specify options like `density`, `colorspace`, `profile`, etc.?

**Comment by snibgo** [comment]
For IM v7, I recommend using "magick". Not "convert" and not "magick convert", which would give you the older v6 syntax. v6 syntax is less powerful, and has less error-checking.
